waiter/src/waiter/core.clj

```diff
@@ -88,14 +88,16 @@
                      "service-id" :service-id-handler-fn
                      "settings" :display-settings-handler-fn
                      "sim" :sim-request-handler
-                     "state" {"" :state-all-handler-fn
-                              "/kv-store" :state-kv-store-handler-fn
-                              "/leader" :state-leader-handler-fn
-                              "/maintainer" :state-maintainer-handler-fn
-                              "/router-metrics" :state-router-metrics-handler-fn
-                              "/scheduler" :state-scheduler-handler-fn
-                              "/statsd" :state-statsd-handler-fn
-                              ["/" :service-id] :state-service-handler-fn}
+                     "state" (array-map ;; maintain key order, explicitly use array-map when size > 8
```

Comment:
@dposada @pschorf @sradack this is where the unit tests for route matching came in handy!

Reply:
I don't think it makes sense to use maps (whether array-map or hash-map) for a data structure where the order of entries is important. I think this is especially true since "maintaining key order" isn't part of the [array-map's public contract](https://clojure.github.io/clojure/clojure.core-api.html#clojure.core/array-map).

These route/pattern entries should *all* be vectors of pairs, not maps. I think that using vectors will make it much more intuitive to anyone reading the code that order matters, and eliminates the need for extra comments trying to clarify that fact. (I already tested this, and bidi is happy with a vector-of-vectors here instead of an array-map).

Reply:
From the [clojure.org reference](https://clojure.org/reference/data_structures#ArrayMaps) 

> When doing code form manipulation it is often desirable to have a map which maintains key order. An array map is such a map - it is simply implemented as an array of key val key val…